app/js/main.js

```diff
@@ -122,3 +122,15 @@ const loadSvg = (r) => {
     nav.init()
     startingBlocks.boot()
 })()
+
+if ('serviceWorker' in navigator) {
+    window.addEventListener('load', function () {
+        navigator.serviceWorker.register('/sw.js').then(function (registration) {
```

Comment:
Service worker path should not be hard coded.

Use a global var and generate this URL from twig.

Reply:
```js
if (window.temp.swPath) {
    if ('serviceWorker' in navigator) {
        console.debug('Navigator support ServiceWorker')
        window.addEventListener('load', function () {
            navigator.serviceWorker.register(window.temp.swPath).then(function (registration) {
                // Registration was successful
                console.log('ServiceWorker registration successful with scope: ', registration.scope)
            }, function (err) {
                // registration failed :(
                console.log('ServiceWorker registration failed: ', err)
            })
        })
    } else if (!window.isSecureContext) {
        console.debug(
            'Navigator cannot init ServiceWorker, you should have a HTTPS secure context.',
            'Or force it unsafe chrome://flags/#unsafely-treat-insecure-origin-as-secure'
        )
    }
}
```